### PR TITLE
Feature/endorse teammates

### PR DESCRIPTION
**Pull request by nuzurie**

### CODE REVIEW

File: Review/delivery/http/review_delivery.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		return
+	}
+
+	loggedID, _ := c.Get("loggedID")
+	logged, _ := loggedID.(string)
+	if logged != reviewer {
+		c.JSON(http.StatusUnauthorized, errors.NewUnauthorizedError("not allowed to get reviews by others"))
+	}
+
+	var student domain.Student
```
Comment (aminebechrifa) [inline Review/delivery/http/review_delivery.go]: to be removed , not used later

File: Review/delivery/http/review_delivery.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	if err != nil || review.Reviewed.ID == "" || review.Tags == nil {
+		fmt.Println(review, err.Error())
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("invalid review body"))
+		return
+	}
+
+	ctx := c.Request.Context()
+
+	var student domain.Student
+	err = c.ShouldBindJSON(&student)
```
Comment (aminebechrifa) [inline Review/delivery/http/review_delivery.go]: not used later

File: Review/delivery/http/review_delivery.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+func (h *ReviewHandler) GetReviewsBy(c *gin.Context) {
+	reviewer := c.Param("reviewer")
+	if reviewer == "" {
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("must provide reviewer id"))
+		return
+	}
+
+	loggedID, _ := c.Get("loggedID")
+	logged, _ := loggedID.(string)
+	if logged != reviewer {
```
Comment (aminebechrifa) [inline Review/delivery/http/review_delivery.go:105]: return after this

File: Review/delivery/http/review_delivery.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		c.JSON(http.StatusUnauthorized, errors.NewUnauthorizedError("not allowed to get reviews by others"))
+	}
+
+	if loggedID != reviewer {
```
Comment (aminebechrifa) [inline Review/delivery/http/review_delivery.go:115]: return after this

**Review by aminebechrifa — CHANGES_REQUESTED** [review]

File: Review/delivery/http/review_delivery.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	var review domain.Review
+	err := c.ShouldBindJSON(&review)
+	if err != nil || review.Reviewed.ID == "" || review.Tags == nil {
+		fmt.Println(review, err.Error())
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("invalid review body"))
+		return
+	}
+
+	loggedID, _ := c.Get("loggedID")
+	reviewer, _ := loggedID.(string)
```
Comment (nuzurie) [inline Review/delivery/http/review_delivery.go:39]: Bug: make sure the loggedID is the same as the reviewer's ID

File: Review/delivery/http/review_delivery.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	if err != nil || review.Reviewed.ID == "" || review.Tags == nil {
+		fmt.Println(review, err.Error())
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("invalid review body"))
+		return
+	}
+
+	ctx := c.Request.Context()
+
+	loggedID, _ := c.Get("loggedID")
+	reviewer, _ := loggedID.(string)
```
Comment (nuzurie) [inline Review/delivery/http/review_delivery.go:77]: set the reviewer's ID in the review

File: Review/delivery/http/review_delivery.go [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+func (h *ReviewHandler) AddReview(c *gin.Context) {
+	reviewed := c.Param("reviewed")
+	if reviewed == "" {
+		c.JSON(http.StatusBadRequest, errors.NewBadRequestError("must provide reviewed id"))
+		return
+	}
+
+	var review domain.Review
+	err := c.ShouldBindJSON(&review)
+	if err != nil || review.Reviewed.ID == "" || review.Tags == nil {
```
Comment (nuzurie) [inline Review/delivery/http/review_delivery.go:32]: Make this conditional inline

**Review by car12lin12 — APPROVED** [review]
Nice job

**Review by stellakimnguyen — APPROVED** [review]

**Pull request merged**